test/parallel/test-repl-external-module.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const { execSync } = require('child_process');
+
+console.log(process.execPath);
+
+const result = execSync(process.execPath, {
+  env: {
+    ...process.env,
+    NODE_EXTERNAL_REPL_MODULE: require.resolve(
+      '../fixtures/external-repl-module.js'),
+  },
+});
+
+assert.strictEqual(result.toString(), '42\n');
```

Comment:
@nodejs/testing anyone know why this is an empty string instead of 42? I use this env var on builds of node on my laptop but it always fails in this test :(

Reply:
I don't much care either way, but it does seem like people prefer `encoding: 'utf-8'` in the `child_process` method options instead of getting a `Buffer` object and calling `toString()`.

Reply:
getting an empty string from this either way... as far as i can tell the other file never even runs.

Reply:
The problem is using `execSync()` with the `node` binary, but I don't exactly know why. If you use `stdio: 'inherit'`, then you'll at least see the `42` in your console, so you know it's running. (Also confirmed with `--inspect-brk` and debugging.) But as soon as you try to capture it with `stdio: 'pipe'` (the default), it kinda just vanishes.

Reply:
Tired and going on vague recollections, but is it probable that when `NODE_EXTERNAL_REPL_MODULE` is being processed, the `node` process is early enough in initialization that it is using `process._rawDebug()` or whatever and not using the stdout one might ordinarily expect, and that's resulting in the output no being captured by `execSync()`? I'm sure I'm messing up all the details....

Reply:
@devsnek I got the test to work by moving it to `test/pseudo-tty` and adding a `test-repl-external-module.out` file that just has `42` in it. I had to slightly modify the test (adding `stdio: 'inherit'` and removing `assert` and `console.log()` from the test file):

```javascript
'use strict';

require('../common');
const fixtures = require('../common/fixtures');
const { execSync } = require('child_process');

execSync(process.execPath, {
  env: {
    ...process.env,
    NODE_EXTERNAL_REPL_MODULE:
      fixtures.path('../fixtures/external-repl-module.js'),
  }, stdio: 'inherit'
});
```

Then, to run it: `tools/test.py test/pseudo-tty/test-repl-external-module.js`